map/framework.cpp

```diff
@@ -2980,3 +2982,26 @@ bool Framework::OriginalFeatureHasDefaultName(FeatureID const & fid) const
 {
   return osm::Editor::Instance().OriginalFeatureHasDefaultName(fid);
 }
+
+bool Framework::HasRouteAltitude() const { return m_routingSession.HasRouteAltitude(); }
+
+bool Framework::GenerateRouteAltitudeChart(size_t width, size_t height, bool day,
+                                           vector<uint8_t> & imageRGBAData) const
+{
+  feature::TAltitudes altitudes;
+  if (!m_routingSession.GetRouteAltitudes(altitudes))
+    return false;
+  deque<double> segDistanceM;
+  if (!m_routingSession.GetSegDistanceM(segDistanceM) && segDistanceM.empty())
+    return false;
+  segDistanceM.push_front(0.0);
```

Comment:
Thanks for the reference from the standard. But when I  said "Amortized complexity" I meant amortized complexity in this case. How much it costs to fill segDistanceM? And how this bound changes when you prepend a single element to the vector? Still O(N), right?